### PR TITLE
make reductions more SYCL friendly

### PR DESCRIPTION
**Pull request by jcosborn**

### CODE REVIEW

**Comment by mathiaswagner** [comment]
Jenkins: Can one of the admins verify this patch?

**Comment by mathiaswagner** [comment]
@Jenkin add to whitelist.

**Comment by mathiaswagner** [comment]
@Jenkins add to whitelist.

File: include/kernels/transform_reduce.cuh [hunk 1]
```diff
@@ -38,6 +38,7 @@ namespace quda {
   template <typename Arg> struct transform_reducer {
     using count_t = decltype(Arg::n_items);
     using reduce_t = decltype(Arg::init_value);
+    using reducer_t = typename Arg::reducer;
```
Comment (maddyscientist) [inline include/kernels/transform_reduce.cuh:41]: Why is this addition needed here?

File: include/targets/cuda/math_helper.cuh [hunk 2]
```diff
@@ -5,6 +5,14 @@
 
 namespace quda {
 
+  /**
+   * @brief Maximum of two numbers
+   * @param a first number
+   * @param b second number
+   */
+  template<typename T>
```
Comment (maddyscientist) [inline include/targets/cuda/math_helper.cuh:13]: * can you add a matching `quda::min` function and add support for `array`?

Comment (maddyscientist) [reply include/targets/cuda/math_helper.cuh:13]: 20eb054

File: include/kernels/transform_reduce.cuh (hunk 1 above)
Comment (jcosborn) [reply include/kernels/transform_reduce.cuh:41]: So I can overload on the reducer type here:
https://github.com/lattice/quda/blob/feature/sycl/include/targets/sycl/reduce_helper.h#L464

**Review by maddyscientist — APPROVED** [review]
